src/upload.js

```diff
@@ -33,10 +32,11 @@ async function upload(pingId, payload) {
                 case response.status >= 200 && response.status < 300:
                     console.info(`Ping submitted successfully ${pingId}`);
                     _deletePersistedPing(pingId);
+                    break;
                 // Unrecoverable error case
                 case response.status >= 400 && response.status < 500:
                     console.error(`Unrecoverable error while submitting ping ${pingId}. Status code: ${response.status}`);
-                    _onUploadUnrecoverableError();
```

Comment:
Wow, this function should exist... This is a bug.

Reply:
Would you mind adding the same contents in line 43, to this deleted line instead?
```js
setTimeout(() => upload(pingId, payload), RECOVERABLE_UPLOAD_ERROR_TIMEOUT);
```
Github doesn't support suggestions for deleted lines.

Reply:
What should it do? Let's do this as a follow-up?

Reply:
Sure.